src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java

```diff
@@ -958,34 +769,22 @@ private int getChars(char[] result) {
                 }
                 // decExponent has 1, 2, or 3, digits
                 if (e <= 9) {
-                    result[i++] = (char) (e + '0');
+                    result[i++] = (byte) (e + '0');
                 } else if (e <= 99) {
-                    result[i++] = (char) (e / 10 + '0');
-                    result[i++] = (char) (e % 10 + '0');
+                    DecimalDigits.putPairLatin1(result, i, e);
+                    i += 2;
                 } else {
-                    result[i++] = (char) (e / 100 + '0');
+                    result[i++] = (byte) (e / 100 + '0');
                     e %= 100;
-                    result[i++] = (char) (e / 10 + '0');
-                    result[i++] = (char) (e % 10 + '0');
+                    DecimalDigits.putPairLatin1(result, i, e);
+                    i += 2;
                 }
             }
             return i;
         }
 
     }
 
-    private static final ThreadLocal<BinaryToASCIIBuffer> threadLocalBinaryToASCIIBuffer =
```

Comment:
Why was this thread local buffer removed?

Reply:
Creating a BinaryToASCIIConverter is not expensive and does not require the use of cache, but this should be proven by performance testing.

Reply:
Each `BinaryToASCIIBuffer` allocates an array for digits. That might be why the buffer was added. @rgiulietti might decide if this removal is fine or if we should keep it.